gcs/src/gcs_act_proto.hpp

```diff
@@ -26,8 +26,10 @@ typedef uint8_t gcs_proto_t;
  *     (needs protocol version bump to keep it identical on all nodes)
  * 4 - fix for the error voting protocol
  *     (must keep it identical on all nodes)
+ * 5 - fix for commit cut tracking for just SYNCED nodes
+ *     (must keep it identical on all nodes)
  */
-#define GCS_PROTO_MAX 4
+#define GCS_PROTO_MAX 5
```

Comment:
so this is creating another incompatibility

Enforced at
`gcs/src/gcs_act_proto.cpp:    if (gu_unlikely(GCS_PROTO_MAX < version)) return -EPROTONOSUPPORT;`

Reply:
This is necessary to know when all nodes in a cluster at upgrade can use new way for calculating commit cut tracking. This commit cut tracking fix is necessary to fix rolling-upgrade bug identified at upgrade testing.

Reply:
So this is because all nodes independently calculate this commit cut and a discrepancy from an old version will result in that node behaving in a way the other node don't expect?

Reply:
Yes